SWIFTLINT.md

```diff
@@ -0,0 +1,138 @@
+# Swift Style Guide
+
+The SwiftLint configuration on this repo is based on the great work of the raywenderlich.com folks [The Official raywenderlich.com Swift Style Guide](https://github.com/raywenderlich/swift-style-guide).
+
+The purpose of this guide is to provide a base format for our Swift codebase. This
+guide describes what are the rules enforced by our swift lint rules and should be 
+used when contributing or reviewing code. 
+
+These guides use SwiftLint as a standard. You can learn more about SwiftLint by visiting its [official documentation page](https://github.com/realm/SwiftLint).
+
+## Table of Contents
+
+* [Installing SwiftLint](#installing-swiftlint)
+* [Using the configuration file](#using-the-configuration-file)
+* [Xcode settings](#xcode-settings)
+* [Running SwiftLint](#running-swiftlint)
+* [Handling rule exceptions](#handling-rule-exceptions)
+* [Approved exceptions](#approved-exceptions)
+	* [Implicitly Unwrapped Optionals](#implicitly-unwrapped-optionals)
+	* [Force cast](#force-cast)
+	* [Force unwrapping](#force-unwrapping)
+	* [SwiftUI and multiple trailing closures](#swiftui-and-multiple-trailing-closures)
+	* [Open-source files](#open-source-files)
+* [Other notes](#other-notes)
+
+## Installing SwiftLint
+
+
+```bash
+brew install swiftlint
```

Comment:
Can we add a line for installing Homebrew?  If someone doesn't have it installed yet, then this command will fail.  I see the note about Homebrew below, but someone following the instructions from top to bottom might miss that.

Reply:
There's one small time bomb hiding here.  Because SwiftLint isn't managed as a versioned dependency, updates to the package can break our build over time.

This happened to me last year—they added a new check which was turned on by default and it failed my build on my CI server until I made a code change to correct the issue.

We avoid this issue on Android, because our linters are usually part of the Gradle build script and are therefore versioned.

I'm not sure there's a good solution to this problem... But at some point we'll probably hit this.  I suppose we could specify a version explicitly?  Or we could just add a note indicating this is a known possible issue?

Reply:
The script that executes SwiftLint in the project can contain a version check.